packages/ember-metal/lib/mixin.js

```diff
@@ -349,7 +346,8 @@ function updateObserversAndListeners(obj, key, observerOrListener, pathsKey, upd
 }
 
 function replaceObserversAndListeners(obj, key, observerOrListener) {
-  var prev = obj[key];
+  var desc = Object.getOwnPropertyDescriptor(obj, key);
```

Comment:
We can handle this more performently, but might require either extra branching or a re-think